kernel/aster-nix/src/fs/named_pipe.rs

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use super::{
+    file_handle::FileLike,
+    pipe::{PipeReader, PipeWriter, DEFAULT_PIPE_BUF_SIZE},
+    utils::{AccessMode, Channel, Metadata, StatusFlags},
+};
+use crate::{
+    events::IoEvents,
+    prelude::*,
+    process::signal::{Pollable, Poller},
+};
+
+pub struct NamedPipe {
+    reader: Arc<PipeReader>,
+    writer: Arc<PipeWriter>,
+}
+
+impl Default for NamedPipe {
+    fn default() -> Self {
```

Comment:
For convenience, add a `new` method to `NamedPipe`, which is same as the `Default` implementation; and for completeness, add an `with_capacity` method so that the user can specify the buffer size of a named pipe.

For consistency, rename `Channel::new` to `Channel::with_capacity`. Similarly, add two functions to `pipe` module: `new_pair` and `new_pair_with_capacity`.